### DISCUSSION TITLE
UnboundLocalError: local variable 'all_sys' referenced before assignment

### DISCUSSION THREAD

**Discussion by wangxinxiny** [deepmodeling/dpgen#742] [Q&A]
Hello, when I run the "nohup dpgen init_bulk param.json machine.json 1>log 2>error &amp;" command, I encountered a problem. The following is my error report. I hope you can help me.
This is the error file：
![image](https://user-images.githubusercontent.com/102158587/170937233-1d0ce421-b144-4fbb-b924-7e6f645da533.png)
This is the dpgen.log file：
![image](https://user-images.githubusercontent.com/102158587/170938090-765daee8-4cc9-4647-a697-59d692ef9d14.png)
and I have also tried to manually convert the OUTCAR to data format, but there are still errors as follows:
![image](https://user-images.githubusercontent.com/102158587/170938878-d0ff5182-79b8-4f97-8549-c94950e88be8.png)

**Comment by AnguseZhang** [comment]
Could you please provide INCAR for ab initio molecular dynamics by VASP? 
Please check `NWRITE` in your POTCAR, which can be refered to https://www.vasp.at/wiki/index.php/NWRITE . 

This tag determines how much will be written to the file [OUTCAR](https://www.vasp.at/wiki/index.php/OUTCAR) ('verbosity tag'). 

You should set `NWRITE` to 2, not 0 or 1. Otherwise, no box and force information will be recorded, such that `dpdata` could not collect information used for training.

**Reply by wangxinxiny** [reply]
Thank you. I have solved this problem according to your tips, but now there is a new problem!
When a dpdata software is used to extract the outcar information of AIMD, it is found that it does not contain virial stress information, but the outcar of single point energy is calculated, which contains virial stress information.
I need to remove dpdata/systems.py to read the code of virial data. How should I do it? I hope you can help me.

**Comment by HuangJiameng** [comment]
"remove dpdata/systems.py to read the code of virial data." What do you mean by that? I can't get your point. :)
